pkg/odo/cli/config/set.go

```diff
@@ -100,21 +111,42 @@ func (o *SetOptions) Run() (err error) {
 	}
 
 	if !o.configForceFlag {
-		if isSet := o.lci.IsSet(o.paramName); isSet {
+		if isSet := o.LocalConfigInfo.IsSet(o.paramName); isSet {
 			if !ui.Proceed(fmt.Sprintf("%v is already set. Do you want to override it in the config", o.paramName)) {
 				fmt.Println("Aborted by the user.")
 				return nil
 			}
 		}
 	}
 
-	err = o.lci.SetConfiguration(strings.ToLower(o.paramName), o.paramValue)
+	err = o.LocalConfigInfo.SetConfiguration(strings.ToLower(o.paramName), o.paramValue)
 	if err != nil {
 		return err
 	}
 
 	log.Info("Local config was successfully updated.")
-	log.Info("Run `odo push --config` command to apply changes to the cluster.")
+	if o.now {
+		o.Context, _, err = genericclioptions.UpdatedContext(o.Context)
+		if o.LocalConfigInfo == nil {
+			fmt.Println("Ooops local config is nil")
+		}
+		glog.V(4).Infof("Reloaded context info %#v" , o)
+
+		if err != nil {
+			return errors.Wrap(err, "unable to retrieve updated local config")
+		}
+		err = o.SetSourceInfo()
+		if err != nil {
+			return errors.Wrap(err, "unable to set source information")
+		}
+		err = o.Push()
```

Comment:
a thought here, if someone does `odo config set --now ABC x ` should we assume they want to essentially do `odo push --config` because well they just changed the config only? but this assumption means that they wouldn't have done any code changes?

Reply:
makes sense, let me see what i can do

Reply:
> a thought here, if someone does `odo config set --now ABC x ` should we assume they want to essentially do `odo push --config` because well they just changed the config only? but this assumption means that they wouldn't have done any code changes?

Nope, we can't assume that. As there might be a code change prior to `odo config set`. In that case `odo config set --now` needs to be as smart as `odo push` - push files if needed.